### ISSUE TITLE
DataTable security-scroll-gradient overlaps the scrollbar

### ISSUE THREAD

**Issue by wdiu** [carbon-design-system/ibm-security#618]
## Bug -
The DataTable security-scroll-gradient overlaps the vertical scrollbar, making it difficult to see the scrollbar.

**Expected behavior -**
Scrollbar should be visible so that users know there is more content which can be accessed by scrolling.

**Actual behavior -**
Can't see the scrollbar clearly.

### Steps for reproducing
1. Go to https://ibm-security.carbondesignsystem.com/?path=/story/components-datatable--default
2. Locate the bx--data-table-container and adjust the width and height to 200px

Observe the scrollbar on the right is overlapped by the gradient and difficult to see.

### Screenshots

![SecurityScrollGradient](https://user-images.githubusercontent.com/10282242/84804477-d502a480-afd0-11ea-970e-2831b62fe2f2.png)

### Affected browsers

Google Chrome Version 83.0.4103.106 (Official Build) (64-bit), Windows 10

### Optional information

**Version -** @carbon/ibm-security v1.26.0-prerelease.8

**Comment by SimonFinney** [comment]
Hi @wdiu - Thanks for raising! Would you mind describing the use case here in detail, so we can better provide a fix?

**Comment by wdiu** [comment]
Hi @SimonFinney, thanks for looking into it.  The issue that I encountered was when I have a DataTable with many columns, it requires horizontal scrolling.  (I just set it to 200 x 200 in the example so it could be reproduced with fewer columns, but in my actual usage of it, the table is of course much wider.)  However, when horizontal scrolling is required, the vertical scrollbar (highlighted in red in the screenshot) becomes very hard to see in both color themes.  Although it becomes clear to the user there is more content horizontally, the issue is that the user does not know they have to scroll down for more content because they cannot see the vertical scrollbar.

If the right gradient didn't overlap the vertical scrollbar, it would be easy to see the vertical scrollbar.

**Comment by SimonFinney** [comment]
@petervachon - would you have some guidance here on how this should behave? Should the scrollbar appear over the gradient?

**Comment by petervachon** [comment]
My first impression is that the scrollbar should appear over the gradient. I'm wondering, what is the use case to set the table height to something fixed? I'd imagine you'd want it to be 100% and if there is scrolling it would be taken care of by the page scrollbar.

**Comment by wdiu** [comment]
Hi @petervachon, thanks for your reply.  In the real application, the table width and height aren't constant at 200px!  In the real application, I have some filters on the left and a few sections (including the DataTable) vertically positioned on the right, so the page scrolling won't be good because I don't want the other sections to scroll too.

It looks something like this:

Left | Right
------------ | -------------
Filters | Content1
Filters | DataTable
Filters | Content2

The only reason I set them to be fixed at 200px was to make it easy to see the issue in the example!

**Comment by petervachon** [comment]
Hi @wdiu thank you for clarifying! That makes sense. 

@SimonFinney I agree, the scrollbar should appear over the gradient

**Comment by SimonFinney** [comment]
@diego-codes I'm wondering if you had guidance or thoughts on this, as you've experience building out the original implementation.

As I understand it, currently the scrollbar exists underneath the gradient as it's defined by the content itself - the overflow on the container is also required, as it ensures that the gradient can be correctly applied to the outer container, the end result being it will always be rendered under the gradient.

Is there an opportunity here to restructure how the `ScrollGradient` is rendered? I imagine you had a lot to consider during the original implementation and would like to understand any challenges or considerations that were surfaced.

**Comment by diego-codes** [comment]
From my understanding, the purpose of the scroll gradient is to demonstrate that there is scrollable content vertically or horizontally. It was never considered that the scrollable element would be scrollable in both directions. 

My original implementation attempt was to use a box-shadow for the gradients, but it didn't seem to work well for me because the box-shadow is multidirectional so you'd end up with gradients on all edges 😭. I think that a box-shadow is the only way to get the gradient underneath the scrollbar, but I'm not 100% sure. I definitely encourage restructuring the `ScrollGradient` if necessary.

**Comment by wdiu** [comment]
Hi @Yohanna, I'm working on a few different projects at the moment, but I was wondering if this still affects the project you're working on?

**Comment by Yohanna** [comment]
Hi @wdiu, I don't believe so.

**Comment by SimonFinney** [comment]
Moving to blocked while confirming whether the `ScrollGradient` should still remain as part of the `DataTable` or conform to Carbon's standards.

/cc @cameroncalder @dbrugger
